native/acter/src/api/push/notification_item.rs

```diff
@@ -303,154 +173,24 @@ pub enum NotificationItemInner {
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-        content: Option<MsgContent>,
-    },
-    Reference {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-    },
-    Comment {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-        content: TextMessageEventContent,
-    },
-    Reaction {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-        key: String,
-    },
-    Creation {
-        parent_obj: NotificationItemParent,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-    },
-    TitleChange {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-    },
-    DescriptionChange {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-        content: Option<TextMessageEventContent>,
-    },
-    // event specific
-    EventDateChange {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-        new_date: UtcDateTime,
-    },
-    // event specific
-    Rsvp {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-        rsvp: RsvpStatus,
-    },
-    // tasks and task list specific
-    TaskAdd {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-    },
-    TaskProgress {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-        done: bool,
-    },
-    TaskDueDateChange {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-        new_due_date: Option<NaiveDate>,
-    },
-    TaskAccept {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-    },
-    TaskDecline {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-    },
-    // catch-all for other object changes
-    OtherChanges {
-        parent_obj: Option<NotificationItemParent>,
-        parent_id: OwnedEventId,
-        room_id: OwnedRoomId,
-        event_id: OwnedEventId,
-    },
+    Activity(Activity),
```

Comment:
Anything we can understand as an object activity is now wrapped in here, coming from the common internal API based on the actual inner object